swap/src/alice/swap.rs

```diff
@@ -502,22 +502,22 @@ impl Swap {
                         let refund_tx_seen =
                             bitcoin_wallet_clone.watch_for_raw_transaction(tx_refund.txid());
 
-                        pin_mut!(punish_tx_finalised);
-                        pin_mut!(refund_tx_seen);
-
-                        state = match select(punish_tx_finalised, refund_tx_seen).await {
-                            Either::Left(_) => AliceState::Punished,
-                            Either::Right((published_refund_tx, _)) => {
+                        select! {
```

Comment:
You can actually return the state from `select!` and assign it to the variable outside. 
Did you implement it with state being set inside on purpose?

Reply:
I was convinced of the opposite. Thanks!